spec.bs

```diff
@@ -491,12 +494,243 @@ Each {{URLPattern}} object has an associated <dfn for=URLPattern>hash component<
+        1. Run [=process a tokenizing error with default position=] given |tokenizer|.
+        1. [=Continue=].
+      1. Let |escaped index| be |tokenizer|'s [=tokenizer/next index=].
+      1. Run [=get the next code point=] given |tokenizer|.
+      1. Run [=add a token with default length=] given |tokenizer|, "[=token/type/escaped-char=]", |tokenizer|'s [=tokenizer/next index=], and |escaped index|.
+
+      1. [=Continue=].
+    1. If |tokenizer|'s [=tokenizer/code point=] is U+007B (`{`):
+      1. Run [=add a token with default position and length=] given |tokenizer| and "[=token/type/open=]".
+      1. [=Continue=].
+    1. If |tokenizer|'s [=tokenizer/code point=] is U+007D (`}`):
+      1. Run [=add a token with default position and length=] given |tokenizer| and "[=token/type/close=]".
+      1. [=Continue=].
+    1. If |tokenizer|'s [=tokenizer/code point=] is U+003A (`:`):
+      1. Let |name position| be |tokenizer|'s [=tokenizer/next index=].
+      1. Let |name start| be |name position|.
+      1. While |name position| is less than |input length|:
+        1. Run [=seek and get the next code point=] given |tokenizer| and |name position|.
+        1. Let |first code point| be true if |name position| equals |name start| and false otherwise.
+        1. Let |valid code point| be the result of running [=is a valid name code point=] given |tokenizer|'s [=tokenizer/code point=] and |first code point|.
+        1. If |valid code point| is false [=break=].
+        1. Set |name position| to |tokenizer|'s [=tokenizer/next index=].
+      1. If |name position| is less than or equal to |name start|:
+        1. Run [=process a tokenizing error=] given |tokenizer|, |name start|, and |tokenizer's| [=tokenizer/index=].
+        1. [=Continue=].
+      1. Run [=add a token with default length=] given |tokenizer|, "[=token/type/name=]", |name position|, and |name start|.
+    1. If |tokenizer|'s [=tokenizer/code point=] is U+0028 (`(`):
+      1. Let |depth| be 1.
+      1. Let |regex position| be |tokenizer|'s [=tokenizer/next index=].
+      1. Let |regex start| be |regex position|.
+      1. Let |error| be false.
+      1. While |regex position| is less than |input length|:
+        1. Run [=seek and get the next code point=] given |tokenizer| and |regex position|.
+        1. If the result of running [=is ASCII=] given |tokenizer|'s [=tokenizer/code point=] is false:
+          1. Run [=process a tokenizing error=] given |tokenizer|, |regex start|, and |tokenizer|'s [=tokenizer/index=].
+          1. Set |error| to true.
+          1. [=Break=].
+        1. If |regex position| equals |regex start| and |tokenizer|'s [=tokenizer/code point=] is U+003F (`?`):
+          1. Run [=process a tokenizing error=] given |tokenizer|, |regex start|, and |tokenizer|'s [=tokenizer/index=].
+          1. Set |error| to true.
+          1. [=Break=].
+        1. If |tokenizer|'s [=tokenizer/code point=] is U+005C (&bpos;):
```

Comment:
I think this should be `\` not `&bpos;`?

Reply:
Bikeshed formatting broke when I tried to use `\`.  Not sure how to put a backslash literal in.

Reply:
I switched to `<code>\</code>` which seems to work.